src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -159,6 +171,17 @@ public NSUrlSessionHandler (NSUrlSessionConfiguration configuration)
 			inflightRequests = new Dictionary<NSUrlSessionTask, InflightData> ();
 		}
 
+#if !MONOMAC
+		void BackgoundNotificationCb (NSNotification obj)
+		{
+			// we do not need to call the lock, we call cancel on the source, that will trigger all the needed code to 
+			// clean the resources and such
+			foreach (var pair in inflightRequests) {
```

Comment:
You can iterate the values, should be more efficiently than asking the `.Value` property for each `pair`